src/shared/dissect-image.c

```diff
@@ -2135,7 +2135,7 @@ int dissected_image_mount(
          *  -EIDRM        → File system is not among allowlisted "common" file systems
          */
 
-        if (!where && (flags & (DISSECT_IMAGE_VALIDATE_OS|DISSECT_IMAGE_VALIDATE_OS_EXT)) != 0)
+        if (!where && FLAGS_SET(flags, DISSECT_IMAGE_VALIDATE_OS|DISSECT_IMAGE_VALIDATE_OS_EXT))
```

Comment:
this is wrong.

FLAGS_SET() checks if *all* specified flags are set. the old code checked if *any* of the specified flags are set